packages/amplify-category-api/src/provider-utils/awscloudformation/utils/containers/set-existing-secret-arns.ts

```diff
@@ -0,0 +1,48 @@
+import { $TSAny, JSONUtilities, pathManager } from 'amplify-cli-core';
+import _ from 'lodash';
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import { category } from '../../../../category-constants';
+import { cfnFileName } from '../containers-artifacts';
+/**
+ * Check if the template contains existing secret configuration and if so, add it to the secretsMap
+ * The secrets configuration is stored in the template in the following format
+ * {
+ *   "Resources": {
+      "TaskDefinition": {
+        "Type": "AWS::ECS::TaskDefinition",
+        "Properties": {
+          "ContainerDefinitions": [
+            {
+              "Secrets": [
+                {
+                  "Name": "SECRETNAME",
+                  "ValueFrom": "<some secrets manager arn>"
+                }
+              }
+            }
+          ]
+        }
+      }
+    }
+  */
+export const setExistingSecretArns = (secretsMap: Map<string, string>, resourceName: string) => {
+  const cfnPath = path.join(pathManager.getBackendDirPath(), category, resourceName, cfnFileName(resourceName));
```

Comment:
Could we make the function accept the parsed CFN and then just set the secrets and move out the loading code into the caller.